server/src/com/cloud/api/dispatch/ParamProcessWorker.java

```diff
@@ -294,6 +294,14 @@ private void setFieldValue(final Field field, final BaseCmd cmdObj, final Object
                     field.set(cmdObj, Float.valueOf(paramObj.toString()));
                 }
                 break;
+            case DOUBLE:
+                // Assuming that the parameters have been checked for required before now,
+                // we ignore blank or null values and defer to the command to set a default
+                // value for optional parameters ...
+                if (paramObj != null && isNotBlank(paramObj.toString())) {
```

Comment:
Consider using ``com.google.common.base.Strings#isNullorEmpty`` method which consolidates the null and blank checks.